plugins/catalog-react/src/components/EntityTable/columns.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Entity,
+  EntityName,
+  RELATION_OWNED_BY,
+  RELATION_PART_OF,
+} from '@backstage/catalog-model';
+import { TableColumn } from '@backstage/core';
+import React from 'react';
+import { getEntityRelations } from '../../utils';
+import {
+  EntityRefLink,
+  EntityRefLinks,
+  formatEntityRefTitle,
+} from '../EntityRefLink';
+
+export function createEntityRefColumn<T extends Entity>({
+  defaultKind,
+}: {
+  defaultKind?: string;
+}): TableColumn<T> {
+  function formatContent(entity: T): string {
+    return formatEntityRefTitle(entity, {
+      defaultKind,
+    });
+  }
+
+  return {
+    title: 'Name',
+    highlight: true,
+    customFilterAndSearch(filter, entity) {
+      // TODO: We could implement this more efficiently, like searching over
+      // each field that is displayed individually (kind, namespace, name).
+      // but that migth confuse the user as it will behave different than a
+      // simple text search.
+      // Another altnerative would be to cache the values. But writing them
+      // into the entity feels bad too.
+      return formatContent(entity).includes(filter);
```

Comment:
Is `filter` the complete string? I guess we should make a helper function that supports tokenisation (space-splitting) of the filter text and make sure that all of those tokens appear in the input